internal/plugin/connectors/http/generic/oauth/v1/protocol.go

```diff
@@ -0,0 +1,222 @@
+package oauth1protocol
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	gohttp "net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type OAuth1 struct {
+	ConsumerKey     string
+	ConsumerSecret  string
+	Nonce           string
+	Signature       string
+	SignatureMethod string
+	TimeStamp       string
+	Token           string
+	TokenSecret     string
+	Version         string
+}
+
+// Oauth Keys - we set this as a constant to make sure that we always verify
+// this exact key syntax and spelling
+const (
+	oauth_consumer_key     = "oauth_consumer_key"
+	oauth_consumer_secret  = "oauth_consumer_secret"
+	oauth_token            = "oauth_token"
+	oauth_token_secret     = "oauth_token_secret"
+	oauth_nonce            = "oauth_nonce"
+	oauth_signature        = "oauth_signature"
+	oauth_signature_method = "oauth_signature_method"
+	oauth_version          = "oauth_version"
+	oauth_timestamp        = "oauth_timestamp"
+
+	oauth_version_number        = "1.0"
+	oauth_signature_method_type = "HMAC-SHA1"
```

Comment:
I think you should put that comment in here that other modes may need to be supported in the future

Reply:
Yup! Added that to the Changelog, Jake is writing up those issues now.